### DISCUSSION TITLE
is it possible to reuse the apple certificate

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [fastlane/fastlane#19252] [Q&A]
I want to reuse the Apple Distribution certificate(because apple only generate max 3 distribution certificate), fisrt step I get the private key from the certificate:

`    openssl pkcs12 -nocerts -nodes -out key.pem -in Certificates.p12
`

second step, I use the private key to generate a new certificate like this:

```
    openssl aes-256-cbc -k 123456 -in key.pem -out MLD9L5TNVK.p12 -a
    openssl aes-256-cbc -k 123456 -in Certificates.cer -out MLD9L5TNVK.cer -a
```

Then I put the certificate to distribution folder of fastlane match. but now I did not know the certiifcate is valid or not, when I use this certificate in match to package a apple app, shows this error:

    [03:59:50]: There are no local code signing identities found.
    You can run `security find-identity -v -p codesigning ***` to get this output.
    This Stack Overflow thread has more information: https://stackoverflow.com/q/35390072/774.
    (Check in Keychain Access for an expired WWDR certificate: https://stackoverflow.com/a/35409835/774 has more info.)
    [03:59:50]: Setting key partition list... (this can take a minute if there are a lot of keys installed)
    [03:59:50]: security: SecItemCopyMatching: The specified item could not be found in the keychain.
    [03:59:50]: security: SecKeychainItemImport: Unknown format in import.

I read the https://stackoverflow.com/q/35390072/774 but I think it is a different problem. why tell me `There are no local code signing identities found` and  `security: SecItemCopyMatching: The specified item could not be found in the keychain` and `security: SecKeychainItemImport: Unknown format in import`. what should I do to fix it?

**Comment by lucgrabowski** [comment]
Hi, why do you want to generate certificate manually?
You can import profile and cert as written in https://docs.fastlane.tools/actions/match/
If you want to anything manually, remember to decrypt match repo: https://docs.fastlane.tools/advanced/other/#manually-manage-the-fastlane-match-repo
And as written above, could you check state of WWDR certificate just in case?
